src/views/Register/index.tsx

```diff
@@ -1,17 +1,393 @@
+import {
+    useCallback,
+    useMemo,
+    useContext,
+} from 'react';
+import {
+    useNavigate,
+    generatePath,
+} from 'react-router-dom';
+import { isTruthyString } from '@togglecorp/fujs';
+import {
+    useForm,
+    ObjectSchema,
+    requiredStringCondition,
+    emailCondition,
+    getErrorObject,
+    PartialForm,
+    createSubmitHandler,
+} from '@togglecorp/toggle-form';
+
+import RouteContext from '#contexts/route';
 import Page from '#components/Page';
+import TextInput from '#components/TextInput';
+import SelectInput from '#components/SelectInput';
+import Link from '#components/Link';
+import Button from '#components/Button';
+import NonFieldError from '#components/NonFieldError';
+
 import useTranslation from '#hooks/useTranslation';
+import useAlert from '#hooks/useAlert';
+import { resolveToComponent } from '#utils/translation';
+import {
+    isValidCountry,
+    isValidNationalSociety,
+} from '#utils/common';
+import type { GET } from '#types/serverResponse';
+import { LabelValue } from '#types/common';
+import {
+    useRequest,
+    useLazyRequest,
+} from '#utils/restRequest';
 
 import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+// FIXME: Dummy type for organization
+type NationalSocietiesList = 'NTLS' | 'DLGN' | 'SCRT' | 'ICRC' | 'OTHR';
+interface OptionType {
+    value: NationalSocietiesList;
+    label: string;
+}
+
+interface CountryOptionType {
+    id: number;
+    name: string;
+}
+
+const nsKeySelector = (item: LabelValue) => item.value.toString();
+const nsLabelSelector = (item: LabelValue) => item.label;
+
+type FormFields = PartialForm<Omit<GET['register'], 'id' | 'token' | 'expires'>>;
+
+const defaultFormValue: FormFields = {};
+
+const keySelector = (item: OptionType) => item.value;
+const labelSelector = (item: OptionType) => item.label;
+
+const countryKeySelector = (item: CountryOptionType) => item.id;
+const countryLabelSelector = (item: CountryOptionType) => item.name;
+
+// FIXME: Need to make an Api for these options
+const organizationTypeOptions: OptionType[] = [
+    { value: 'NTLS', label: 'National Society' },
+    { value: 'DLGN', label: 'IFRC' },
+    { value: 'SCRT', label: 'ICRC' },
+    { value: 'ICRC', label: 'Other' },
+];
 
 // eslint-disable-next-line import/prefer-default-export
 export function Component() {
     const strings = useTranslation(i18n);
+    const { login: loginRoute } = useContext(RouteContext);
+
+    const formSchema: ObjectSchema<FormFields> = useMemo(
```

Comment:
Let's move this outside the function.
We will handle translation of errors separately.